crates/primitives/src/transaction/mod.rs

```diff
@@ -861,7 +863,54 @@ impl TransactionSigned {
         if num_txes < *PARALLEL_SENDER_RECOVERY_THRESHOLD {
             txes.into_iter().map(|tx| tx.recover_signer()).collect()
         } else {
-            txes.into_par_iter().map(|tx| tx.recover_signer()).collect()
+            let mut recovered_signers: Vec<Address> = Vec::new();
+
+            let chunks = rayon::current_num_threads() * 8;
+            let chunk_size: usize = (num_txes / chunks) / 8;
+            let chunk_size = chunk_size.max(2);
```

Comment:
these constants seem a bit too arbitrary/random

Reply:
Can definitely change that